### PR TITLE
Course Change - Feature flag clean up

### PR DESCRIPTION
**Pull request by JR-G**
## Context

Once the Course Change feature goes live it's time to remove the feature flag and get rid of the previous support UI tool that was used in place of this.

## Changes proposed in this pull request

- Remove `:change_course_details_before_offer`
- Update add course choice support tooling to include all unsuccessful end states
- 🪓  all support UI course change functionality

## Link to Trello card

https://trello.com/c/b4YUsS7V

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/components/support_interface/application_choice_component.rb [hunk 1]
```diff
@@ -181,7 +181,7 @@ def visible_over_register_api?
     end
 
     def change_course_choice_link
-      return {} unless @application_choice.application_form.editable? && ApplicationStateChange::DECISION_PENDING_STATUSES.include?(application_choice.status.to_sym)
+      return {} unless !FeatureFlag.active?(:hide_support_change_course_tool) && @application_choice.application_form.editable? && ApplicationStateChange::DECISION_PENDING_STATUSES.include?(application_choice.status.to_sym)
```
Comment (carlosmartinez) [inline app/components/support_interface/application_choice_component.rb:184]: you can now enjoy the glory of `FeatureFlag.inactive?`

Comment (JR-G) [reply app/components/support_interface/application_choice_component.rb:184]: 🤩

File: app/services/feature_flag.rb [hunk 2]
```diff
@@ -32,7 +32,6 @@ def feature
     [:summer_recruitment_banner, 'Show a banner to indicate a shorter recruitment timeframe during summer', 'Richard Pattinson'],
     [:support_user_revert_withdrawn_offer, 'Allows a support user to revert an application withdrawn by the candidate', 'James Glenn'],
     [:draft_vendor_api_specification, 'The specification for Draft Vendor API v1.1', 'Abeer Salameh'],
-    [:change_course_details_before_offer, 'Allows providers to change course choice details before the point of offer', 'James Glenn'],
```
Comment (cwrw) [inline app/services/feature_flag.rb:35]: could we add a data migration to remove this feature as well from the db?

File: app/models/application_form.rb [hunk 3]
```diff
@@ -304,7 +304,11 @@ def maximum_number_of_course_choices?
   end
 
   def support_cannot_add_course_choice?
-    application_choices.where.not(status: :withdrawn).count >= maximum_number_of_course_choices
+    number_of_unsuccessful_application_choices >= maximum_number_of_course_choices
```
Comment (cwrw) [inline app/models/application_form.rb:307]: could you add a test for this please?

Comment (cwrw) [inline app/models/application_form.rb:307]:
```suggestion
   application_choices.where.not(status: ApplicationStateChange::UNSUCCESSFUL_END_STATES).count >= maximum_number_of_course_choices
```

File: app/services/feature_flag.rb (hunk 2 above)
Comment (JR-G) [reply app/services/feature_flag.rb:35]: 🆗

File: app/controllers/support_interface/application_forms/courses_controller.rb [hunk 4]
```diff
@@ -45,26 +45,6 @@ def create
...
-        @change_course_choice = ChangeCourseChoiceForm.new(change_course_choice_params)
-
-        begin
-          if @change_course_choice.save(params[:application_choice_id])
-            flash[:success] = 'Course successfully changed'
-            redirect_to support_interface_application_form_path(application_form_id)
-          else
-            render :edit
-          end
-        rescue CourseChoiceError, ProviderInterviewError => e
```
Comment (cwrw) [inline app/controllers/support_interface/application_forms/courses_controller.rb:62]: could we remove the course choice and the provider interview error as I don't think we use them anywhere?

**Review by cwrw — COMMENTED** [review]
Nicely and cleanly done! Only comment would be could we remove the `ChangeApplicationChoiceCourseOption` service and spec as we no longer need it?
After that we can merge this early next week ⭐

**Review by cwrw — APPROVED** [review]
🌟

**Pull request merged**